src/components/virtual-scroll/virtual-scroll.ts

```diff
@@ -368,7 +369,13 @@ export class VirtualScroll implements DoCheck, AfterContentInit, OnDestroy {
   /**
    * @input {function} Same as `ngForTrackBy` which can be used on `ngFor`.
    */
-  @Input() virtualTrackBy: TrackByFn;
+  @Input() set virtualTrackBy(val: TrackByFn) {
+    if (!isPresent(val)) return;
+    this._virtualTrackBy = val;
+    if (isPresent(this._records)) {
+      this._differ = this._iterableDiffers.find(this._records).create(this.virtualTrackBy);
```

Comment:
can we move the this._differ creation logic to a new function? maybe `_updateDiffer`. it is duplicated in here and virtualScroll.

Also, can you add a new-line after @Input()?